java/ql/test/utils/model-generator/typebasedflow/p/Stream.java

```diff
@@ -0,0 +1,166 @@
+package p;
+
+import java.util.*;
+import java.util.function.*;
+import java.util.stream.*;
+
+public class Stream<T> {
```

Comment:
Something's weird here. We're importing `java.util.stream.*` such that the standard `Stream` class is in scope, but we're also defining a class with the same name, which I guess shadows the imported name?? Are we attempting to stub-implement `java.util.stream.Stream` or import it?

Reply:
I wonder then why the generated models use `Element`? I mean they certainly ought to for `java.util.stream.Stream` but this stub class isn't that. If the stub class was implementing, say, `Iterable<T>` then it would make more sense, as that ought to be enough to make it use `Element` as opposed to generating a synthetic field.

Reply:
When we're stubbing like this, then I think we should use explicit imports from `java.util.stream` such that we aren't shadowing anything. That would make it more clear that the references to `Stream` are in fact this stub class.

Reply:
(1) As far as can tell from the documentation of the `Stream` (java) interface and `ContainerType` (QL) implementation, then `Stream` will not be identified as a `ContainerType` (note that `Stream<T>` does not implement `Iterable<T>`). The implementation on this PR will also perceive classes or interfaces having any method that returns an `Iterator` as being a container like type (but this purely has effect the type based model generation implementation. 
(2) Imports has been changed to avoid confusion.